scripts/irods/test/test_iput.py

```diff
@@ -886,6 +887,50 @@ def test_large_put(self):
         self.object_name = 'test_large_put'
         self.run_tests(self.object_name, 5120000)
 
+    def test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947(self):
+        self.object_name = 'test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947'
+        irodsctl = IrodsController()
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        # create a 100 MiB file
+        file_size = 100*1024*1024
```

Comment:
I have not been able to reproduce the error.  I think it requires a sufficiently fast network to an S3 implementation that also generates the "reduce your request rate" error.

I have seen that error in AWS but don't remember what scenario created it.  When I try with a very small buffer size to AWS it does a lot of read request but not at a rate that would trigger this error.  It just takes a long time to read the full file.  It would probably require some type of parallel request which we don't do in this instance.

I also tried with minio but I don't think it generates that type of error.

But to answer the question, in general I can put a 100 MiB file to S3 with checksum without creating the error.